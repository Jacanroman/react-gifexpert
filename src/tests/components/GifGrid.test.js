import React from 'react';
import { shallow } from 'enzyme';
import GifGrid from '../../components/GifGrid';
import {useFetchGifs} from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs'); //simularrrr useFetchGifs

describe('pruebas con <GifGrid />', () => {

    const category = 'goku'

    
    
    test('debe mostrar <GifGrid /> correctamente',()=>{

        useFetchGifs.mockReturnValue({
            data:[],
            loading:true
        })

        const wrapper = shallow(<GifGrid category={category} />)
        expect(wrapper).toMatchSnapshot()

    })

    test('debe de mostrar items cuando se cargan imagenes useFetchGifs',()=>{

        const gifs = [{
            id:'ABC',
            url:'https://localhost',
            title: 'something'
        },
        {
            id:'134',
            url:'https://localhost',
            title: 'something'
        }]


        useFetchGifs.mockReturnValue({
            data:gifs,
            loading:false
        })

        const wrapper = shallow(<GifGrid category={category} />)
        
        
        expect(wrapper).toMatchSnapshot()
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length)
    })

})
