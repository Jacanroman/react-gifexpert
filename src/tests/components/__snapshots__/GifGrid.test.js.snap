// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas con <GifGrid /> debe de mostrar items cuando se cargan imagenes useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    goku
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="something"
      url="https://localhost"
    />
    <GifGridItem
      id="134"
      key="134"
      title="something"
      url="https://localhost"
    />
  </div>
</Fragment>
`;

exports[`pruebas con <GifGrid /> debe mostrar <GifGrid /> correctamente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    goku
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
